src/common/buffer.cc

```diff
@@ -1826,7 +1826,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (!_buffers.empty()) {
       ptr &l = _buffers.back();
       if (l.get_raw() == bp.get_raw() &&
-	  l.end() == bp.start() + off) {
+	  l.end() == off) {
```

Comment:
the `off` here is the offset inside of `bp` which start from `_off` in the backed raw data, while `l.end()` is the offset from the raw data. i don't think they are comparable.